golang/cosmos/x/swingset/keeper/snapshotter.go

```diff
@@ -365,6 +380,21 @@ func (snapshotter *SwingsetSnapshotter) RestoreExtension(height uint64, format u
 		return snapshots.ErrUnknownFormat
 	}
 
+	err := snapshotter.checkNotActive()
+	if err != nil {
+		return err
+	}
+
+	active := &activeSnapshot{
+		isRestore: true,
+		height:    int64(height),
+		logger:    snapshotter.logger,
+	}
```

Comment:
Initialize the `done` channel, and make sure this path is covered by a test.

Reply:
I don't believe creating a `done` channel is required, as it's only used in `select` with `default` cases.
A test would have to do something unsupported, which is to start `RestoreExtension` in a goroutine, then call `InitiateSnapshot` from the main thread (or inversely)